app/services/tagging_history/index_page.rb

```diff
@@ -0,0 +1,35 @@
+module TaggingHistory
+  class IndexPage
+    attr_reader :link_changes, :filter_by_user_options
+
+    def initialize(params = { link_types: ['taxons'] })
+      @filter_by_user_options = {}
+      @link_changes = get_link_changes(params)
+    end
+
+  private
+
+    def get_link_changes(params)
+      results = Services.link_changes_api
+                  .get_link_changes(params)
+                  .to_hash
+                  .deep_symbolize_keys[:link_changes]
+
+      results.each do |result|
+        user = User.find_by_uid(result[:user_uid])
+        next unless user.present?
+        result[:user_uid] = user.uid
+        result[:user_name] = user.name
+        @filter_by_user_options[user.name] = user.uid
+        result[:organisation] = user.organisation_slug.try do |slug|
+          slug.capitalize.tr('-', ' ')
+        end
+      end
+
+      # The Publishing API returns the results ordered so that the
+      # latest come last. Reverse this so that the latest events are
+      # displayed first.
+      results.reverse
```

Comment:
you could also iterate using reverse_each instead of each, which saves making a copy of the array and may be slightly more efficient

Reply:
I put this in before reversing the order that the publishing-api was returning the results in, so I've removed it now as its unnecessary.